integration/tests/cook/test_basic.py

```diff
@@ -997,23 +997,56 @@ def check_unique_constraint(unscheduled_jobs):
     def test_balanced_host_constraint(self):
```

Comment:
Change requested: Can we split this test into two tests, one that runs for a small cluster (`minimum_hosts == num_hosts + 1`) and one that runs for a larger cluster.